test/e2e/serial/tests/tolerations.go

```diff
@@ -783,10 +827,51 @@ func sriovToleration() corev1.Toleration {
 	}
 }
 
-func waitForMcpUpdate(cli client.Client, ctx context.Context, mcps []*machineconfigv1.MachineConfigPool) {
-	By("waiting for mcp to start updating")
-	Expect(waitForMcpsCondition(cli, ctx, mcps, machineconfigv1.MachineConfigPoolUpdating)).To(Succeed())
+func waitForMcpUpdate(cli client.Client, ctx context.Context, mcpsInfo []mcpInfo, updateType MCPUpdateType) {
+	for _, info := range mcpsInfo {
+		By(fmt.Sprintf("verify updates for mcp %q", info.obj.Name))
+		klog.Info("waiting for mcp to start updating")
+		if err := waitForMcpsCondition(cli, ctx, []*machineconfigv1.MachineConfigPool{info.obj}, machineconfigv1.MachineConfigPoolUpdating); err != nil {
+			klog.Warningf("failed to find mcps while in updating status")
+		}
```

Comment:
I'd drop this part. Checking we updated to the desired state should be sufficient

Reply:
I don't think that would be right in case the node didn't flip faster than the automation. additionally we check only a sample/random node, we need at least some waiting time before making sure the mcp got updated so we stick with a meaningful wait on the updating state